src/Atmos/Model/moisture.jl

```diff
@@ -45,89 +57,44 @@ Assumes the moisture components are computed via thermodynamic equilibrium.
-  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
-  air_pressure(phase)
-end
-function soundspeed(::EquilMoist, state::Vars, aux::Vars, t::Real)
-  ρ = state.ρ
-  q_tot = state.ρq_tot / ρ
-  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
-  soundspeed_air(phase)
-end
+function gradvariables!(m::EquilMoist, transform::Vars, state::Vars, aux::Vars, t::Real)
+  invρ = state.ρ
+  transform.moisture.q_tot = state.ρq_tot * invρ
 
-function gradvariables!(::EquilMoist, transform::Vars, state::Vars, aux::Vars, t::Real)
-  ρ = state.ρ
-  q_tot = state.ρq_tot / ρ
-  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
+  phase = thermo_state(m, state, aux)
   q = PhasePartition(phase)
+  R_m = gas_constant_air(q)
+  T = aux.moisture.temperature
+  e_tot = state.ρe * invρ
 
-  transform.moisture.q_vap = q.tot - q.liq - q.ice
-  transform.moisture.q_liq = q.liq
-  transform.moisture.q_ice = q.ice
-  transform.moisture.temperature = aux.moisture.temperature
+  # a better name?
+  transform.moisture.e_x = e_tot + R_m*T
 end
 
 
-function diffusive!(::EquilMoist, diffusive::Vars, ∇transform::Grad, state::Vars, aux::Vars, t::Real, ν::Union{Real,AbstractMatrix})
+function diffusive!(m::EquilMoist, diffusive::Vars, ∇transform::Grad, state::Vars, aux::Vars, t::Real, ν::Union{Real,AbstractMatrix})
   # turbulent Prandtl number
   diag_ν = ν isa Real ? ν : diag(ν) # either a scalar or vector
-  D_q_vap = D_q_liq = D_q_ice = D_q_tot = diag_ν / Prandtl_t # either a scalar or vector
-  
-  # diffusive flux of q_tot
-  # FIXME 
-  ρd_q_vap = state.ρ * (-D_q_vap) .* ∇transform.moisture.q_vap # a vector
-  ρd_q_liq = state.ρ * (-D_q_liq) .* ∇transform.moisture.q_liq # a vector
-  ρd_q_ice = state.ρ * (-D_q_ice) .* ∇transform.moisture.q_ice # a vector
-
-  diffusive.moisture.ρd_q_tot = ρd_q_vap + ρd_q_liq + ρd_q_ice
-
   D_T = diag_ν / Prandtl_t
-  phase = PhaseEquil(aux.moisture.e_int, q_tot, ρ, aux.moisture.temperature)
-  q = PhasePartition(phase)
 
-  # J is the conductive or SGS turbulent flux of sensible heat per unit mass
-  ρJ = state.ρ * cv_m(q) * D_T * ∇transform.moisture.temperature
+  # diffusive flux of q_tot
+  diffusive.moisture.ρd_q_tot = state.ρ * (-D_T) .* ∇transform.moisture.q_tot
 
-  # D is the total specific energy flux
-  T = aux.moisture.temperature
-  u = state.ρu / state.ρ
-
-  # FIXME
-  I_vap = cv_v * (T - T_0) + e_int_v0
-  I_liq = cv_l * (T - T_0)
-  I_ice = cv_v * (T - T_0) - e_int_i0
-  e_kin = 0.5 * sum(abs2,u)
-  e_pot = grav * aux.coord.z
-  e_tot_vap = e_kin + e_pot + I_vap
-  e_tot_liq = e_kin + e_pot + I_liq
-  e_tot_ice = e_kin + e_pot + I_ice
-  ρD = state.ρ * ((e_tot_vap + R_v*T)*diffusive.moisture.ρd_q_vap + e_tot_liq*diffusive.moisture.ρd_q_liq + e_tot_ice*diffusive.moisture.ρd_q_ice) 
-
-  diffusive.moisture.ρJ_ρD = ρJ + ρD
+  # diffusive flux of total energy
+  diffusive.moisture.ρJ_ρD = state.ρ * (-D_T) .* ∇transform.transform.moisture.e_x
```

Comment:
@tapios is this what you had in mind?

Reply:
Yes, looks good. I'd just give `\rho J_\rho D' a more intuitive name, e.g., `\rho_SGS_energyflux` or something like it. I also do not find the `e_x` very clear (subscript `x` is confusing--for us it usually means the zonal coordinate)